### DISCUSSION TITLE
Fuzzy search + gin index

### DISCUSSION THREAD

**Discussion by zidkim** [supabase/supabase#12097] [Questions]
Trying to implement fuzzy search with `pg_trgm` with an index.
I'm essentially trying to continue from the solution shared on this [thread](https://github.com/supabase/supabase/discussions/5435#discussioncomment-2172376).

With the following query, I believe a proper gin index has been created on the `fts` column, however, I'm having trouble creating the appropriate fuzzy search query using the `fts` column. That said, the issue(s) could be more complicated than my assumptions.

create index column and add gin index
```
alter table
  books
add column
  fts tsvector generated always as (to_tsvector('english', title || ' ' || japanese_title)) stored;

create index books_idx on books using gin (fts); 

select id, fts
from books;
```

I expect something like the following to work, but can't seem to get it right. Notice the `b.fts`.
```
CREATE OR REPLACE FUNCTION public.search_books(book_term varchar)
RETURNS table(id int, title text, author text)
LANGUAGE plpgsql
AS $$
	begin
		return query
			SELECT b.id, b.title, b.author 
			FROM books b 
			WHERE book_term % ANY(STRING_TO_ARRAY(b.fts, ' '));
	end;
$$;
```

Many thanks in advance.

**Comment by Ali-Aizaz** [comment]
use one of these functions in pg_trgm 
![Screenshot 2024-01-06 225616](https://github.com/supabase/supabase/assets/50923380/2b732422-0d64-432d-8262-80173b435b2d)

https://www.postgresql.org/docs/current/pgtrgm.html
